src/hardware/setup.rs

```diff
@@ -255,24 +258,19 @@ pub fn setup(
     let rcc = device.RCC.constrain();
     let ccdr = rcc
         .use_hse(8.mhz())
-        .sysclk(400.mhz())
+        .sysclk(design_parameters::SYSCLK)
         .hclk(200.mhz())
-        .per_ck(100.mhz())
+        .per_ck(design_parameters::TIMER_FREQUENCY)
         .pll2_p_ck(100.mhz())
         .pll2_q_ck(100.mhz())
         .freeze(vos, &device.SYSCFG);
 
     // Before being able to call any code in ITCM, load that code from flash.
     load_itcm();
 
-    // Set up the system timer for RTIC scheduling.
-    {
-        let tim15 =
-            device
-                .TIM15
-                .timer(10.khz(), ccdr.peripheral.TIM15, &ccdr.clocks);
-        system_timer::SystemTimer::initialize(tim15);
-    }
+    let systick = Systick::new(core.SYST, ccdr.clocks.sysclk().0);
+
+    core.SCB.enable_icache();
```

Comment:
Is there any particular reason we're enabling the ICACHE earlier? Just to speed up INIT? If there's an important reason, we should document it. Otherwise, this is fine.

Reply:
It moved around a couple times. No particular reason. It should just be after the ITCM load.